src/ImageSharp/Formats/DecoderOptions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System;
+
+namespace SixLabors.ImageSharp.Formats
+{
+    /// <summary>
+    /// Provides general configuration options for decoding image formats.
+    /// </summary>
+    public sealed class DecoderOptions
+    {
+        private static readonly Lazy<DecoderOptions> LazyOptions = new(() => new());
+
+        private uint maxFrames = int.MaxValue;
+
+        /// <summary>
+        /// Gets the shared default general decoder options instance.
+        /// </summary>
+        public static DecoderOptions Default { get; } = LazyOptions.Value;
+
+        /// <summary>
+        /// Gets or sets a custom Configuration instance to be used by the image processing pipeline.
+        /// </summary>
+        public Configuration Configuration { get; set; } = Configuration.Default;
+
+        /// <summary>
+        /// Gets or sets the target size to decode the image into.
+        /// </summary>
+        public Size? TargetSize { get; set; } = null;
```

Comment:
It looks like we are missing tests validating various usages of `TargetSize` (decode to specific size & keep aspect ratio). These tests can be also used as codes sample to ask some community members to do a final sanity-check on the target size decoding API.

Reply:
You mean validating the result of given dimensions? I would have thought that would already be covered by `ResizeHelperTests`? We simply pass the size to that and `JpegDecoderCore` already has tests for scaled decoding.

Reply:
I mean end-to-end testing resize-decode the way users will work with the feature, including aspect-ratio preservation. The only setter call I see for `DecoderOptions.TargetSize` is in `LoadResizeSaveStressRunner`.

Even though currently it's implemented with a simple `Resize`, it's a separate feature that needs testing.  Later it could be optimized by utilizing resizer internals (feeding the image to `ResizeWorker` row-by-row).

Reply:
> I only see unit tests for SpectralConverter.CalculateResultingImageSize. Or am I missing something?

Yep, I've left actual image tests to the moment when API is ready as jpeg resize-decode entry point was somewhat of a stub. As that was internal we didn't risk exposing invalid code to users but this PR is a game changer. I know this solution isn't 'synchronized' but I'll write proper test suite for jpeg after API is set in stone.

Reply:
I'm fine delaying it for jpeg, the implementation is somewhat special anyways, but we could add test for another decoder (eg. PNG) in this PR. Then later we can extend as we optimize decoders.